src/core/border_router/routing_manager.cpp

```diff
@@ -3689,55 +3707,76 @@ void RoutingManager::PdPrefixManager::ProcessPlatformGeneratedRa(const uint8_t *
     }
 }
 
-Error RoutingManager::PdPrefixManager::Process(const RouterAdvert::RxMessage &aMessage)
+void RoutingManager::PdPrefixManager::ProcessDhcpPdPrefix(const PrefixTableEntry &aPrefixTableEntry)
+{
+    Error error = kErrorInvalidState;
+
+    if (mEnabled)
```

Comment:
In `ProcessPlatformGeneratedRa()` we have ` VerifyOrExit(IsRunning()...` check. But here we check `mEnabled`.
These seem to be two different bools corresponding to different states in `PdPrefixManager`.
- We have both `mEnabled` and `mIsRunning`.
- Should we check `IsRunning()` here as well?